### PR TITLE
[Integration][DBT] Support for CLL in DBT integration

### PR DESCRIPTION
**Pull request by mayurmadnani**
### Problem

DBT integration does not have CLL support

Closes: #3260 

### Solution

DBT gives table level lineage in manifest. Using the SQL parser and dbt compiled code, CLL is generated.

#### One-line summary:

Adds CLL support using openlineage SQL parser to derive column lineage

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] Your comment includes a one-liner for the changelog about the specific purpose of the change (_not required for changes to tests, docs, or CI config_)
- [x] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

**Comment by mobuchowski** [comment]
Thanks for the contribution @mayurmadnani. It would be great to add tests that check validity of the solution here. The tests are located here: https://github.com/OpenLineage/OpenLineage/tree/0399867fc3cf29015c7de1f8923b49ec0b263617/integration/common/tests/dbt

**Comment by mayurmadnani** [comment]
Thanks @mobuchowski 
Added tests. Also, here is a sample run in local mode

image

**Comment by JDarDagran** [comment]
@mayurmadnani do you have knowledge about how compiled SQL is created in results file? Do database object always stay the same in SQL statement or do they have database / schema added if they're missing in the model definition?

On the other hand, maybe this is very rare case because usually one does refer to db objects using {{ ref }}?

**Comment by mayurmadnani** [comment]
> @mayurmadnani do you have knowledge about how compiled SQL is created in results file? Do database object always stay the same in SQL statement or do they have database / schema added if they're missing in the model definition?
> 
> On the other hand, maybe this is very rare case because usually one does refer to db objects using {{ ref }}?

@JDarDagran when you use {{ ref }} dbt will compile and generate the compiled SQL with db and schema info. see this- 
image

**Comment by JDarDagran** [comment]
Is this the same for {{ source }} and tests? Sorry for being too inquisitive, just trying to make sure we don't miss some common use case.
I think your contribution is great 🚀

**Comment by mayurmadnani** [comment]
> Is this the same for {{ source }} and tests? Sorry for being too inquisitive, just trying to make sure we don't miss some common use case. I think your contribution is great 🚀

Yes, holds for source and tests as well
image
image

**Comment by JDarDagran** [comment]
Awesome, thanks for checking

File: integration/common/tests/dbt/test_dbt_local.py [hunk 1]
```diff
@@ -268,3 +268,88 @@ def test_build_target_path(test_name, dbt_project, expected):
...
+        project_dir=path,
+    )
+    processor.dbt_run_metadata = parent_run_metadata
+    dbt_events = processor.parse()
+    events = [
+        attr.asdict(event, value_serializer=serialize)
+        for event in dbt_events.starts + dbt_events.completes + dbt_events.fails
+    ]
+    with open(f"{path}/result.json") as f:
+        assert match(json.load(f), events)
```
Comment (mobuchowski) [inline integration/common/tests/dbt/test_dbt_local.py:295]: @mayurmadnani there is no column-level lineage in the result event: https://github.com/OpenLineage/OpenLineage/blob/main/integration/common/tests/dbt/compiled_code/result.json

The comparison won't check existence of CLL in the event your change is generating.

**Review by mobuchowski — CHANGES_REQUESTED** [review]

**Pull request closed without merging**